app/src/main/java/com/example/sharingang/UserProfileFragment.kt

```diff
@@ -220,23 +226,23 @@ class UserProfileFragment : Fragment() {
     }
 
     private fun setupReportButton() {
-        var username = ""
-        if (currentUserId != null) {
-            binding.btnReport.visibility = View.VISIBLE
-            lifecycleScope.launch(Dispatchers.IO) {
-                username = userRepository.get(shownUserProfileId!!)!!.name
+        if(currentUserId != null) {
+            lifecycleScope.launch(Dispatchers.Main) {
+                val hasBeenReported =
+                    userRepository.hasBeenReported(currentUserId!!, shownUserProfileId!!)
```

Comment:
A user can report him/her-self, I don't think one should be able to do that

Reply:
For now I don't know what else I can do, because I have not found a method for logging in two different users during testing, so this would probably give me ~20% coverage overall. Any suggestions?